packages/api-explorer/src/utils/hooks.spec.ts

```diff
@@ -60,19 +60,14 @@ describe('Navigate', () => {
     })
   })
 
-  test('clears existing params when given params are an empty object', () => {
```

Comment:
Wanted to discuss if we should still account for the case of an empty object. We currently use `undefined` as a value to indicate we are keeping the key that already exists in the URL, and `null` to indicate that we are getting rid of parameters. If you look at the way I changed the code, individual parameters can be removed, as well as all of them, if passing in `null` entirely. 

Let me know thoughts or we can discuss in call.

Reply:
I'm ok with this. @jkaster @bryans99 thoughts?

I don't see tests unit that cover this though.
1. Unit test to show the additiveness. i.e if current query params are `foo=foo` and I pass in `bar=bar` then I end up with both.
2. Unit test to show that if current params are `foo=foo` and `bar=bar`, passing in `bar=null` should leave me with `foo=foo`.
3. Unit test to show that passing in null clears all existing query params

Reply:
@josephaxisa See new tests, also I thought the test starting in line 56 (clears existing params when given params are null) addressed your 3rd point.